### PR TITLE
Fix errors for phpstan level 6

### LINKED ISSUE #658: Increase PHPStan to level 6
Todo:

- [ ] Increase PHPStan configuration to level 6
- [ ] Resolve all issues

### PR DESCRIPTION
**Pull request by j3j5**
Hi, I tried to fix all errors from phpstan on level 6. Not sure if you have a specific wish for the format for iterable types (`array<int,string>` vs `string[]`) but for the rest it should be fine. 

Actually, I think I've found a bug on the implementation of strip-hosts, but please double check that part before merging just in case I misunderstood how it's supposed to work.

It should fix #658

### CODE REVIEW

File: src/Builder/ArchiveBuilderHelper.php [hunk 1]
```diff
@@ -87,6 +99,12 @@ protected function isOneOfNamesInList(array $names, array $list): bool
         return false;
     }
 
+    /**
+     *
+     * @param string $name
```
Comment (GromNaN) [inline src/Builder/ArchiveBuilderHelper.php:104]: Some `@param` and `@return` annotations are redundant with method signature; is it really required by phpstan?

Comment (alcohol) [reply src/Builder/ArchiveBuilderHelper.php:104]: I don't believe so.

These was a [discussion about the superfluous phpdoc rule](https://github.com/composer/satis/discussions/679); but so far consensus was to keep the rule in place.

Comment (dereuromark) [reply src/Builder/ArchiveBuilderHelper.php:104]: It is also more consistent for the time being.
And to be honest:
Removing some and adding others is usually counterproductive in readability, it really confuses people as to the positioning of those lines then if they ever need to be added.

**Comment by alcohol** [comment]
That merge commit was not what I intended to do. Let me see if I can resolve this from CLI.

**Comment by alcohol** [comment]
See #731

**Pull request closed without merging**